libkbfs/block_tree.go

```diff
@@ -505,6 +519,63 @@ func (bt *blockTree) getBlocksForOffsetRange(ctx context.Context,
 		return offsetI.Less(offsetJ)
 	})
 
+	// If we are returning data even though not all the goroutines completed,
+	// we may need to return only some of the data we gathered in order to
+	// return a correct prefix of the data. Thus, we find the longest prefix of
+	// the data without any holes.
+	if mustCheckForHoles {
+		var prevPath []parentBlockAndChildIndex
+		for pathIdx, path := range pathsFromRoot {
+			// First path should be at startOff. Each subsequent path should
+			// immediately follow the preceding path.
+			pathIsCorrect := true
+			if pathIdx == 0 {
+				lastBlockInPath := path[len(path)-1]
+				_, off := lastBlockInPath.pblock.IndirectPtr(
+					lastBlockInPath.childIndex)
+				if off != startOff {
```

Comment:
Hrm, this isn't true actually. startOff just has to be somewhere within the given block, not necessarily the first byte.  (For example, the block could start at byte 0 and be 1024 bytes long, but the read could start at byte 100.)

For that matter, we have this concept of "holes" in file data, since you can "truncate" a file to a very large size and it doesn't make sense to write all those zeroes.  So the first block could legitimately have an offset that is greater than `off`.  Bleh.

Maybe each job and result should track whether this is the left-most requested block for the request overall.  So the first job would have some bit set, and it would transfer that bit to the first job it requests, and so on.  And when a block or error is returned when the bit is set for that job also sets the bit in the result.  Then if the above code gets a `DeadlineExceeded` error and the result bit is set, you can just return the error since you know you didn't get any valid data.  (And this check here goes away.)